subsys/mgmt/suitfu/src/suitfu_mgmt_suit_image_fetch.c

```diff
@@ -228,29 +227,26 @@ int suitfu_mgmt_suit_missing_image_upload(struct smp_streamer *ctx)
 		offset_in_image += req.img_data.len;
 	}
 
-	component_lock();
 	if (session->transfer_completed_cvar && session->uri && session->uri_length &&
 	    session->session_id == req.stream_session_id) {
 		if (rc != 0) {
 			session->return_code = SUIT_PLAT_ERR_IO;
 			k_condvar_signal(session->transfer_completed_cvar);
 			component_unlock();
 			image_size = 0;
+			component_unlock();
```

Comment:
Double unlock seems not needed, line 235 already contains one.

Reply:
Nice catch, fixed!